Source/WebCore/style/StyleAdjuster.cpp

```diff
@@ -984,6 +985,16 @@ void Adjuster::adjustForSiteSpecificQuirks(RenderStyle& style) const
     }
 #endif
 #endif
+
+    if (m_document->quirks().needsYouTubeDarkModeQuirk()) {
+        // Sets color-scheme to dark if dark attribute is applied to root element.
+        if (RefPtr div = dynamicDowncast<HTMLHtmlElement>(m_element)) {
+            static MainThreadNeverDestroyed<const AtomString> darkMode("dark"_s);
+            if (div->hasAttribute(darkMode))
```

Comment:
```suggestion
        if (m_element && m_element == m_document->documentElement()) {
            static MainThreadNeverDestroyed<const AtomString> darkMode("dark"_s);
            if (m_element->hasAttribute(darkMode))
```

This is probably slightly more performant.

Reply:
Fixed.